File: api/clients/src/main/java/com/foodagram/clients/content/dto/RecipeUpdateDto.java
package com.foodagram.clients.content.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecipeUpdateDto {
    private UUID id;
    private UUID postId;
    private String title;
    private String description;
    private List<IngredientUpdateDto> ingredients;
    private List<String> instructions;
    private String cuisineType;
    private String difficultyLevel;
} 

File: admin-panel/foodagram-admin-panel/src/models/content/dto/RecipeUpdateDto.ts
import {IngredientUpdateDto} from './IngredientUpdateDto';

export interface RecipeUpdateDto {
    title?: string;
    description?: string;
    ingredients?: IngredientUpdateDto[];
    instructions?: string[];
    preparationTime?: number;
    cookingTime?: number;
    servings?: number;
    difficulty?: string;
    cuisine?: string;
    mealType?: string;
    dietaryRestrictions?: string[];
    prepTime: number;
}
